v1.0/sql-faqs.md

```diff
@@ -86,6 +86,36 @@ Not at this time, but storing a 16-byte array in a [`BYTES`](bytes.html) column
 
 When an [`ORDER BY`](select.html#sorting-retrieved-values) clause is not used in a `SELECT` query, retrieved rows are not sorted by any consistent criteria. Instead, CockroachDB returns them as the coordinating node receives them.
 
+## Why are my `INT` columns returned as strings in JavaScript?
+
+In CockroachDB, all `INT`s are represented with 64 bits of precision, but JavaScript numbers only have 53 bits of precision. This means that large integers stored in CockroachDB are not exactly representable as JavaScript numbers. For example, JavaScript will round the integer `235191684988928001` to the nearest representable value, `235191684988928000`. Notice that the last digit is different. This is particularly problematic when using the `unique_rowid()` [function](functions-and-operators.html), since `unique_rowid()` nearly always returns integers that require more than 53 bits of precision to represent.
+
+To avoid this loss of precision, Node's [`pg` driver](https://github.com/brianc/node-postgres) will, by default, return all CockroachDB `INT`s as strings.
+
+~~~ javascript
+// Schema: CREATE TABLE users (id INT DEFAULT unique_rowid(), name STRING);
+pgClient.query("SELECT id FROM users WHERE name = 'Roach' LIMIT 1", function(err, res) {
+  var idString = res.rows[0].id;
+  // idString == '235191684988928001'
+  // typeof idString == 'string'
+});
+~~~
+
+To perform another query using the value of `idString`, you can simply use `idString` directly, even where an `INT` type is expected. Either the Postgres driver or CockroachDB will automatically convert the string to a CockroachDB `INT`.
```

Comment:
what is meant by "Either the Postgres driver or CockroachDB"? shouldn't it always be Cockroach doing this conversion? or does the driver somehow check if the string is within the int bounds?

Reply:
Node's `pg` driver does its own type conversion provided you use prepared queries. So in a query like

```query("UPDATE foo SET int_col = $1", ["1234"])```

the `pg` driver sees an integer type hint on "$1" and will send the integer `1234` over the wire, instead of the string `"1234"`.

The reason we needed string -> number literal coercion in CockroachDB itself was to cope with Sequelize, which eschews prepared queries in favor of string mashing, e.g.:

```
query("UPDATE foo SET int_col = '" + escape("1234") + "'")
```

That's too detailed an explanation, of course, and I don't think we want a string mashing example in our docs. Too easy for people to think that we're encouraging string mashing.

Perhaps we can replace the sentence with "The string will automatically be coerced into a CockroachDB `INT`", and leave out mentioning exactly which layer is doing the conversion?

Reply:
Done.